apps/aecore/lib/aecore/structures/spend_tx.ex

```diff
@@ -3,43 +3,132 @@ defmodule Aecore.Structures.SpendTx do
   Aecore structure of a transaction data.
   """
 
+  @behaviour Aecore.Structures.Transaction
   alias Aecore.Structures.SpendTx
+  alias Aeutil.Serialization
+  alias Aeutil.Parser
+  alias Aecore.Structures.Account
+  alias Aecore.Chain.ChainState
+  alias Aecore.Wallet
 
+  require Logger
+
+  @typedoc "Expected structure for the Spend Transaction"
+  @type payload :: %{
+    to_acc: Wallet.pubkey(),
+    value: non_neg_integer(),
+    lock_time_block: non_neg_integer()
+  }
+
+  @typedoc "Reason for the error"
+  @type reason :: String.t()
+
+  @typedoc "Structure that holds specific transaction info in the chainstate.
+  In the case of SpendTx we don't have a subdomain chainstate."
+  @type tx_type_state() :: %{}
+
+  @typedoc "Structure of the Spend Transaction type"
   @type t :: %SpendTx{
-    from_acc: binary(),
-    to_acc: binary(),
+    to_acc: Wallet.pubkey(),
     value: non_neg_integer(),
-    nonce: non_neg_integer(),
-    fee: non_neg_integer(),
-    lock_time_block: non_neg_integer(),
-    data: map()
+    lock_time_block: non_neg_integer()
   }
 
   @doc """
   Definition of Aecore SpendTx structure
 
   ## Parameters
-  - nonce: A random integer generated on initialisation of a transaction.Must be unique
-  - from_acc: From account is the public address of one account originating the transaction
   - to_acc: To account is the public address of the account receiving the transaction
-  - value: The amount of a transaction
+  - value: The amount of tokens send through the transaction
+  - lock_time_block: In which block the tokens will become available
   """
-  defstruct [:from_acc, :to_acc, :value, :nonce, :fee, :lock_time_block, :data]
+  defstruct [:to_acc, :value, :lock_time_block]
   use ExConstructor
 
-  @spec create(binary(), binary(), non_neg_integer(), non_neg_integer(), non_neg_integer(), non_neg_integer(), map()) :: {:ok, TxData.t()}
-  def create(from_acc, to_acc, value, nonce, fee, lock_time_block \\ 0, data \\ %{}) do
-    {:ok, %SpendTx{from_acc: from_acc,
-                  to_acc: to_acc,
-                  value: value,
-                  nonce: nonce,
-                  fee: fee,
-                  lock_time_block: lock_time_block,
-                  data: data}}
+  # Callbacks
```

Comment:
why have this callbacks comment?

Reply:
Maybe to indicate that they are not just some functions created in the SpendTx but come as callbacks from the Behavior, if you think it's unecessary I'll remove them

Reply:
okay, just asking